web/lib/components/word/Speech.jsx

```diff
@@ -1,26 +1,50 @@
+import React from 'react'
 import { getVoiceSettings } from '../../storage'
 import { VolumnIcon } from '../common/Icons'
 
 const LANG = 'en-GB'
 
+function getVoices() {
+  return new Promise((resolve) => {
+    let interval
+
+    interval = setInterval(() => {
+      const voices = window.speechSynthesis.getVoices()
+      if (voices.length > 0) {
+        clearInterval(interval)
+        resolve(voices)
+      }
+    }, 10)
+  })
+}
+
 export function Speech({ word }) {
-  const doSpeak = () => {
-    const selectedVoice = getVoiceSettings().get()
+  const [voice, setVoice] = React.useState(null)
 
-    const synth = window.speechSynthesis
-    const voice = synth
-      .getVoices()
-      .find((voice) =>
+  React.useEffect(() => {
+    ;(async function () {
+      const voices = await getVoices()
+      const selectedVoice = getVoiceSettings().get()
+
+      const voice = voices.find((voice) =>
         selectedVoice ? voice.voiceURI === selectedVoice : voice.lang === LANG,
       )
 
-    if (voice) {
-      const utterance = new SpeechSynthesisUtterance(word)
-      utterance.lang = voice.lang
-      utterance.voice = voice
-      synth.speak(utterance)
-    }
+      if (voice) {
+        setVoice(voice)
+      }
+    })()
+  }, [])
+
+  const doSpeak = () => {
+    const synth = window.speechSynthesis
+    const utterance = new SpeechSynthesisUtterance(word)
+    utterance.lang = voice.lang
+    utterance.voice = voice
+    synth.speak(utterance)
   }
```

Comment:
Because value change in `localStorage` doesn't trigger a re-render,  `voice` in this function will stuck with its initial value. We need to call `getVoiceSettings().get()` and find the selected voice inside `doSpeak`.